src/tools/perf/lib/libperf.c

```diff
@@ -1219,14 +1219,18 @@ static ucs_status_t ucp_perf_test_setup_endpoints(ucx_perf_context_t *perf,
     unsigned peer_index  = rte_peer_index(group_size, group_index);
     unsigned i;
 
-    if ((group_size != 2) &&
-            !((group_size == 1) &&
-                        (perf->params.flags & UCX_PERF_TEST_FLAG_LOOPBACK))) {
-        ucs_error("perftest requires group size to be exactly 2 "
+    if ((perf->params.flags & UCX_PERF_TEST_FLAG_LOOPBACK) &&
+                                                   (group_size != 1)) {
+        ucs_error("perftest loopback requires group size to be 1 "
+                  "(actual group size: %u)", group_size);
+        return UCS_ERR_UNSUPPORTED;
+    } else if ((group_size != 2) &&
+                 !(perf->params.flags & UCX_PERF_TEST_FLAG_LOOPBACK)) {
```

Comment:
1. can remove "else" here so it would be 2 separate 'if' checks
2. minor: for consistency, check LOOBPACK flag before group size
3. pls fix code style - align ( on column